### ISSUE TITLE
Massive go-slow in XRDP after tigervnc upgrade from 1.13 to 1.14

### ISSUE THREAD

**Issue by nixuser58** [TigerVNC/tigervnc#1810]
**Describe the bug**
I am using XRDP on a number of Fedora 40 systems, all using tigervnc as the Xvnc backend to XRDP.  Up to tigervnc-1.13.1-14.fc40 this worked well.  The upgrade to tigervnc-1.14.0-2.fc40 has made the connection so slow it is unusable over a LAN.   Reverting to the previous version of tigervnc fixes the problem.  Moving away from tigervnc as an XRDP backend to an xorgxrdp backend also fixes the problem.

**To Reproduce**
1. Update a working Fedora 40 system to tigervnc-1.14.0-2.fc40
2. XRDP becomes unusable

**Client (please complete the following information):**
 - OS:  Linux Fedora 40 currently kernel-6.10.5-200.fc40.x86_64
 - VNC client:  remmina-1.4.33-1.fc40.x86_64 used as an XRDP client
 - Client downloaded from:  Fedora

**Server (please complete the following information):**
 - OS: Linux Fedora 40 currently kernel-6.10.5-200.fc40.x86_64
 - VNC server: tigervnc-1.14.0-2.fc40 and xrdp-0.10.1-1.fc40.x86_64
 - Server downloaded from: Fedora 40
 - Server was started using: Starts via xrdp as the [Xvnc] backend.

**Additional context**
I am not having this problem on all servers.  Out of 5 XRDP servers the problem is happening on 2.  

The common thread on the failing systems is both systems are legacy boxes (Dell Precision T3600  & Dell Precision T5610) and both are running legacy NVIDIA cards with the default nouveau drivers.  Also, both GPUs are from the kepler generation but this may be a coincidence.

The working systems in my case are all running Intel graphics of mixed generations (all working fine after upgrading to tigervnc-1.14.0-2.fc40).

**Comment by CendioOssman** [comment]
Can you test a plain TigerVNC server on this system? The Xrdp setup is rather complex and difficult to determine where an issue originates.

Which desktop environment are you using?

**Comment by nixuser58** [comment]
This has become a bit of a mess because the platform I have access to at the moment has changed.  I reported this on Fedora 40 (release version) but the platform is now Fedora 41 beta branch (but it is the same physical system previously demonstrating the problem).

I installed tigervnc-server (rather than the tigervnc-server-minimal that XRDP requires) then got the server going.  I connected using the TigerVNC 64-bit client exe file from a Win box.  Xfce worked in full speed mode.

I then connected using XRDP and selected the Xvnc backend (I had moved to the xordxrdp backend) and it still worked in full speed mode, so I can no longer demonstrate the problem on this system.

I will have access to the other system in a couple of days that is still running Fedora 40 and I'll report my findings.

**Comment by nixuser58** [comment]
I think we're going to have to close this.  

On the second system, I was unable to get the stand alone TigerVNC server working correctly with the TigerVNC client on another Fedora 40 box.  The connection video was scrambled, like an old CRT based TV that had lost horizontal sync.  I could connect only in full-screen mode, even after hours of trouble shooting.  I can confirm in full-screen mode a VNC only connection was working in the slow mode, but since there was obviously something else wrong I don't think the result is necessarily meaningful.

In the end I updated the test box to the 41 beta branch with reverted settings using XRDP.  I established things worked correctly with an xorgxrdp backend, then reverted to a TigerVNC backend, and after rebooting things were still working correctly.

So whatever it was, it was specific to only two of my boxes, and updating them to Fedora 41 beta fixed them.
